internal/upnp/upnp.go

```diff
@@ -527,6 +527,33 @@ type getExternalIPAddressResponse struct {
 	NewExternalIPAddress string `xml:"NewExternalIPAddress"`
 }
 
+type soapErrorResponseEnvelope struct {
+	XMLName xml.Name
+	Body    soapFaultResponseBody `xml:"Body"`
```

Comment:
For our purposes so far, I think we only need a smaller struct?

```go
type soapErrorResponseEnvelope struct {
   ErrorCode int `xml:"Body>Fault>detail>UPnPError>errorCode"`
   ErrorDescription int `xml:"Body>Fault>detail>UPnPError>errorDescription"`
```

(barring typos)